docs/documentation/streams/quickstart.rst

```diff
@@ -0,0 +1,345 @@
+real-time applications and microservices, where the input and/or output
+data is stored in Kafka clusters. Kafka Streams combines the simplicity
+of writing and deploying standard Java and Scala applications on the
+client side with the benefits of Kafka's server-side cluster technology
+to make these applications highly scalable, elastic, fault-tolerant,
+distributed, and much more.
+
+This quickstart example will demonstrate how to run a streaming
+application coded in this library. Here is the gist of the
+``WordCountDemo`` example code (converted to use Java 8 lambda
+expressions for easy reading).
+
+.. code:: bash
+
+    // Serializers/deserializers (serde) for String and Long types
+    final Serde<String> stringSerde = Serdes.String();
+    final Serde<Long> longSerde = Serdes.Long();
+
+    // Construct a `KStream` from the input topic "streams-plaintext-input", where message values
+    // represent lines of text (for the sake of this example, we ignore whatever may be stored
+    // in the message keys).
+    KStream<String, String> textLines = builder.stream("streams-plaintext-input",
+        Consumed.with(stringSerde, stringSerde);
+
+    KTable<String, Long> wordCounts = textLines
+        // Split each text line, by whitespace, into words.
+        .flatMapValues(value -> Arrays.asList(value.toLowerCase().split("\\W+")))
+
+        // Group the text words as message keys
+        .groupBy((key, value) -> value)
+
+        // Count the occurrences of each word (message key).
+        .count()
+
+    // Store the running counts as a changelog stream to the output topic.
+    wordCounts.toStream().to("streams-wordcount-output", Produced.with(Serdes.String(), Serdes.Long()));
+
+It implements the WordCount algorithm, which computes a word occurrence
+histogram from the input text. However, unlike other WordCount examples
+you might have seen before that operate on bounded data, the WordCount
+demo application behaves slightly differently because it is designed to
+operate on an **infinite, unbounded stream** of data. Similar to the
+bounded variant, it is a stateful algorithm that tracks and updates the
+counts of words. However, since it must assume potentially unbounded
+input data, it will periodically output its current state and results
+while continuing to process more data because it cannot know when it has
+processed "all" the input data.
+
+As the first step, we will start Kafka (unless you already have it
+started) and then we will prepare input data to a Kafka topic, which
+will subsequently be processed by a Kafka Streams application.
+
+`Step 1: Download the code <#quickstart_streams_download>`__
+------------------------------------------------------------
+
+`Download <https://www.apache.org/dyn/closer.cgi?path=/kafka/%7B%7BfullDotVersion%7D%7D/kafka_%7B%7BscalaVersion%7D%7D-%7B%7BfullDotVersion%7D%7D.tgz>`__
+the {{fullDotVersion}} release and un-tar it. Note that there are
```

Comment:
This macros like {{fullDotVersion}} and {{dotVersion}} are only for html. And should be modified in rst.

Reply:
@ewencp ? ^